pkg/reconciler/apibinding/apibinding_reconcile.go

```diff
@@ -0,0 +1,528 @@
+				conditionsv1alpha1.ConditionSeverityWarning,
+				"error getting previously bound APIExport: %v",
+				err,
+			)
+
+			// Stop reconciling, but return error so we retry
+			return reconcileStatusStop, fmt.Errorf("error getting APIExport %s|%s: %w", apiExportClusterName, apiBinding.Spec.Reference.Workspace.ExportName, err)
+		}
+
+		var exportedSchemas []*apisv1alpha1.APIResourceSchema
+		// name -> uid
+		exportedSchemaUIDs := map[string]string{}
+
+		for _, schemaName := range apiExport.Spec.LatestResourceSchemas {
+			apiResourceSchema, err := r.getAPIResourceSchema(apiExportClusterName, schemaName)
+			if err != nil {
+				conditions.MarkFalse(
+					apiBinding,
+					apisv1alpha1.APIExportValid,
+					apisv1alpha1.GetErrorReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					"error getting APIResourceSchema %s|%s: %v",
+					apiExportClusterName,
+					schemaName,
+					err,
+				)
+
+				// No way to proceed from here - don't retry
+				return reconcileStatusStop, nil
+			}
+
+			exportedSchemas = append(exportedSchemas, apiResourceSchema)
+			exportedSchemaUIDs[schemaName] = string(apiResourceSchema.UID)
+		}
+
+		if apiExportLatestResourceSchemasChanged(apiBinding, exportedSchemas) {
+			klog.Infof("APIBinding %s|%s needs rebinding because the APIExport's latestResourceSchemas has changed", apiBinding.ClusterName, apiBinding.Name)
+
+			apiBinding.Status.Phase = apisv1alpha1.APIBindingPhaseBinding
+
+			// Commit the phase change, then wait for next update event to process binding changes
+			return reconcileStatusStop, nil
+		}
+	}
+
+	return reconcileStatusContinue, nil
+}
+
+func getAPIExportClusterName(apiBinding *apisv1alpha1.APIBinding) string {
+	org, _, err := helper.ParseLogicalClusterName(apiBinding.ClusterName)
+	if err != nil {
+		// should never happen
+		klog.Errorf("Error parsing logical cluster %q: %v", apiBinding.ClusterName, err)
+		return ""
+	}
+
+	return helper.EncodeOrganizationAndClusterWorkspace(org, apiBinding.Spec.Reference.Workspace.WorkspaceName)
+}
+
+func referencedAPIExportChanged(apiBinding *apisv1alpha1.APIBinding) bool {
+	// Can't happen because of OpenAPI, but just in case
+	if apiBinding.Spec.Reference.Workspace == nil {
+		return false
+	}
+
+	if apiBinding.Status.BoundAPIExport == nil {
+		return false
+	}
+	if apiBinding.Status.BoundAPIExport.Workspace == nil {
+		return false
+	}
+
+	return *apiBinding.Spec.Reference.Workspace != *apiBinding.Status.BoundAPIExport.Workspace
+}
+
+func apiExportLatestResourceSchemasChanged(apiBinding *apisv1alpha1.APIBinding, exportedSchemas []*apisv1alpha1.APIResourceSchema) bool {
+	exportedSchemaUIDs := sets.NewString()
+	for _, exportedSchema := range exportedSchemas {
+		exportedSchemaUIDs.Insert(string(exportedSchema.UID))
+	}
+
+	boundSchemaUIDs := sets.NewString()
+	for _, boundResource := range apiBinding.Status.BoundResources {
+		boundSchemaUIDs.Insert(boundResource.Schema.UID)
+	}
+
+	return !exportedSchemaUIDs.Equal(boundSchemaUIDs)
+}
```

Comment:
so no hash?

Reply:
No, I decided to just check UIDs. Since APIResourceSchema is immutable, the only way something can change is with a delete & recreate, which means a new UID. Right? Can always think through a better way to do this as a follow up

Reply:
fine for now.